app/lib/state_importer.rb

```diff
@@ -0,0 +1,24 @@
+require 'yaml'
+
+##
+# this module upserts State data
+module StateImporter
+  STATE_YAML = Rails.root.join("db/states.yml")
+
+  ##
+  # output normally except in test
+  def say(message)
+    puts message unless Rails.env.test?
+  end
+
+  def upsert_states(yml = STATE_YAML)
+    say "beginning state upsert using environment: #{Rails.env}"
+    states = YAML.load_file(yml)['states']
+    State.destroy_all
+    states.each do |datum|
+      state = State.find_by(abbreviation: datum['abbreviation'])
+      state.present? ? State.update!(name: datum['name']) : State.create!(abbreviation: datum['abbreviation'], name: datum['name'])
+    end
```

Comment:
Since this will be run on live data as part of the deploy, it should be in a transaction so that there is no possibility of an inconsistent state during the import (and especially if the import fails for some reason).

```ruby
State.transaction do
  State.destroy_all
  states.each do ... end
end
```

Reply:
is it an "upsert" if you destroy_all?

Reply:
+1 very true.